axlearn/cloud/gcp/lws_utils.py

```diff
@@ -0,0 +1,196 @@
+                "projects/<reservation_project>/reservations/<reservation>"
+                https://github.com/GoogleCloudPlatform/ai-on-gke/blob/889ec98f9b9a7aec05eb0f9890ada1f4c59b6159/tpu-provisioner/internal/cloud/gke.go#L328-L334
+        """
+
+        reservation: Optional[str] = None
+        reservation_project: Optional[str] = None
+
+    @classmethod
+    def define_flags(cls, fv: flags.FlagValues):
+        super().define_flags(fv)
+        common_kwargs = dict(flag_values=fv, allow_override=True)
+        flags.DEFINE_string("reservation", None, "TPU reservation.", **common_kwargs)
+        flags.DEFINE_string(
+            "reservation_project", None, "TPU reservation project.", **common_kwargs
+        )
+
+    @classmethod
+    def from_flags(cls, fv: flags.FlagValues, **kwargs) -> Config:
+        cfg: TPULeaderWorkerTemplate.Config = super().from_flags(fv, **kwargs)
+        default_env = get_default_env(
+            tpu_type=infer_tpu_type(fv.instance_type),
+            num_tpu_slices=fv.num_replicas,
+            job_name=cfg.name,
+        )
+        # NOTE: we allow fv.env flags to override the defaults.
+        cfg.env_vars = {**default_env, **cfg.env_vars, **parse_kv_flags(fv.env)}
+        cfg.reservation = cfg.reservation or gcp_settings("gke_reservation", required=False, fv=fv)
+        cfg.reservation_project = cfg.reservation_project or gcp_settings(
+            "gke_reservation_project", required=False, fv=fv
+        )
+        return cfg
+
+    def __init__(self, cfg, *, bundler):
+        super().__init__(cfg, bundler=bundler)
+        cfg: TPULeaderWorkerTemplate.Config = self.config
+        self._tpu_type = infer_tpu_type(cfg.accelerator.instance_type)
+        if self._tpu_type not in USER_FACING_NAME_TO_SYSTEM_CHARACTERISTICS:
+            raise NotImplementedError(f"Missing system characteristics for {self._tpu_type}")
+
+    def _build_container(self) -> dict:
+        """Build the container to be used in the leader template
+
+        Returns:
+            A nested dict correspoding to a k8s Container config.
+        """
+        cfg: TPULeaderWorkerTemplate.Config = self.config
+        system = USER_FACING_NAME_TO_SYSTEM_CHARACTERISTICS[self._tpu_type]
+        env_vars = {**cfg.env_vars}
+        k8s_env_vars = [dict(name=k, value=str(v)) for k, v in env_vars.items()]
+        resources = {"limits": {"google.com/tpu": system.chips_per_vm}}
+
+        return dict(
+            name=cfg.name,
+            image=self._bundler.id(cfg.name),
+            command=["bash", "-c", cfg.command],
+            resources=resources,
+            env=k8s_env_vars,
+            imagePullPolicy="Always",
+        )
+
+    def _build_pod(self) -> dict:
+        cfg: TPULeaderWorkerTemplate.Config = self.config
+        system = USER_FACING_NAME_TO_SYSTEM_CHARACTERISTICS[self._tpu_type]
+        annotations, labels, selector = {}, {}, {}
+        if cfg.reservation is not None:
```

Comment:
```suggestion
        if cfg.reservation:
```
Please use implicit Python boolean, same elsewhere

Reply:
fixed